src/modules/commons/components/overflowMenu/OverflowMenu.js

```diff
@@ -0,0 +1,264 @@
+ * @property {boolean} [disabled] whether or not the menu item is enabled
+ * @property {boolean} [isDivider] whether or not the menu item stands for a divider
+ */
+
+
+const DefaultMenuOption = { label: null, icon: null, action: null, disabled: false, isDivider: false };
+
+/**
+ * @enum
+ */
+export const MenuTypes = Object.freeze({
+	MEATBALL: 'meatball',
+	KEBAB: 'kebab'
+});
+
+/**
+ *
+ * Properties:
+ * - `items`
+  * - `type`
+ *
+ * @class
+ * @author thiloSchlemmer
+ */
+export class OverflowMenu extends MvuElement {
+
+	constructor() {
+		super({
+			type: MenuTypes.MEATBALL,
+			menuItems: [],
+			isCollapsed: true,
+			anchorPosition: null
+		});
+		this._documentListener = { pointerdown: null, pointerup: null };
+	}
+
+	update(type, data, model) {
+		switch (type) {
+			case Update_IsCollapsed:
+				return { ...model, isCollapsed: data };
+			case Update_Menu_Type:
+				return { ...model, type: data };
+			case Update_Menu_Items:
+				return {
+					...model, menuItems: data.map(i => {
+						return { ...DefaultMenuOption, ...i };
+					})
+				};
+			case Update_Anchor_Position:
+				return { ...model, anchorPosition: data };
+		}
+	}
+
+
+	/**
+	 * @override
+	 */
+	createView(model) {
+		const { isCollapsed, type } = model;
+		const onClick = (e) => {
+			if (isCollapsed) {
+				this._registerDocumentListener('pointerdown');
+				this._registerDocumentListener('pointerup');
+			}
+			else {
+				this._deregisterDocumentListener('pointerdown');
+				this._deregisterDocumentListener('pointerup');
+			}
+			const rect = this.getBoundingClientRect();
+			const delta = [e.x - e.offsetX, e.y - e.offsetY];
+			this.signal(Update_Anchor_Position, { absolute: [rect.x, rect.y], relative: [rect.x - delta[0], rect.y - delta[1]] });
+
+			this.signal(Update_IsCollapsed, !isCollapsed);
+		};
+
+		const menu = this._getMenuOrFixedNotification(model);
+		return html`
+		 <style>${css}</style> 
+         <button class='anchor'>
+            <span id="menu-icon" data-test-id class='menu__button ${type} ' @click=${onClick} >
+            </span>
+         </button>	         
+         ${menu}
+     </div>`;
+	}
+
+	_getMenuOrFixedNotification(model) {
+		const { isCollapsed, menuItems, anchorPosition } = model;
+
+		const { EnvironmentService: environmentService } = $injector.inject('EnvironmentService');
+		if (environmentService.isTouch()) {
+			if (isCollapsed) {
+				clearFixedNotification();
+			}
+			else {
+				emitFixedNotification(this._getItems(menuItems));
+			}
+			return nothing;
+		}
+
+		const sector = anchorPosition ? this._calculateSector(anchorPosition.absolute) : 0;
+
+		//consider css arrow offset of 10px
+		const yOffset = (sector < 2 ? 0 : -1) * 35;
+		const xOffset = (sector === 1 || sector === 2 ? 1 : 0) * 35;
+
+		const style = anchorPosition ? { '--anchor-x': anchorPosition.relative[0] + xOffset + 'px', '--anchor-y': anchorPosition.relative[1] + yOffset + 'px' } : {};
```

Comment:
We should explain our approach here, e.g. like in `MapContextMenu`

```
/**
* Correct positioning of the context menu is a bit tricky, because we don't know
* (and can't calculate) the dimensions of the context menu and its content child before rendering.
* Therefore we translate the element after rendering by a css transformation.
*/

```